src/main/java/duke/Storage.java

```diff
@@ -0,0 +1,91 @@
+package duke;
+
+import duke.exception.DukeException;
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.Task;
+import duke.task.Todo;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Scanner;
+
+/**
+ * Deals with loading tasks from the given file and saving tasks in the same file.
+ */
+public class Storage {
+    private File file;
+
+    /**
+     * Constructs a Storage object given the file path.
+     *
+     * @param filePath The file path where the tasks are stored. It can be
+     *                 either relative or absolute.
+     */
+    public Storage(String filePath) {
+        this.file = new File(filePath);
+    }
+
+    /**
+     * Loads the existing tasks from the data file.
+     *
+     * @return An ArrayList of tasks which are stored as serialized string in the data file
+     * @throws DukeException if the file does not exit or the data in the file
+     *                       does not follow the correct format (i.e the serialized form of task objects)
+     */
+    public ArrayList<Task> load() throws DukeException {
+        ArrayList<Task> tasks = new ArrayList<>();
+        try {
+            Scanner scanner = new Scanner(file);
+            while (scanner.hasNextLine()) {
+                String[] data = scanner.nextLine().split(" \\| ");
+                Task task;
+                switch (data[0]) {
+                case "T":
+                    task = new Todo(data[2]);
```

Comment:
I could not find any code that catches Index Out of Bound Exception in the event that the array fails to split properly. Maybe you can catch that Exception too.